app/src/components/common/TmSessionImportName.vue

```diff
@@ -99,29 +123,146 @@ export default {
+        return this.networkHDPaths[this.attempt]
+      } else if (this.currentNetwork.network_type === `polkadot`) {
+        return this.networkCurves[this.attempt]
+      } else {
+        return null
+      }
+    },
+    currentCryptoView() {
+      if (this.currentNetwork.network_type === `cosmos`) {
+        return cryptoDictionary[
+          this.networkHDPaths[this.attempt].replace(/\'/g, "")
+        ]
+      } else if (this.currentNetwork.network_type === `polkadot`) {
+        return cryptoDictionary[this.networkCurves[this.attempt]]
+      } else {
+        return ``
+      }
+    },
   },
-  async created() {
-    this.importedAddress = await this.$store.dispatch(`getAddressFromSeed`, {
-      seedPhrase: this.$store.state.recover.seed,
-      network: this.network,
-    })
+  mounted() {
+    this.created()
   },
   methods: {
     onSubmit() {
       this.$v.$touch()
       if (this.$v.name.$invalid) return
       this.$router.push("/recover/password")
     },
+    currentHDPathOrCurve() {
+      if (this.currentNetwork.network_type === `cosmos`) {
+        return {
+          HDPath: this.currentCrypto,
+          curve: this.networkCurves[0], // ed25519
+        }
+      } else if (this.currentNetwork.network_type === `polkadot`) {
+        return {
+          HDPath: this.networkHDPaths[0], // no clue
+          curve: this.currentCrypto,
+        }
+      } else {
+        return {
+          HDPath: ``,
+          curve: ``,
+        }
+      }
+    },
+    async created(retry = false) {
+      if (retry) this.attempt++
+      this.attempt = this.numberAttemptsController(
+        this.networkCryptoTypes,
+        this.attempt
+      )
+      const wallet = await getWallet(
+        this.$store.state.recover.seed,
+        this.currentNetwork,
+        this.currentHDPathOrCurve()[`HDPath`],
+        this.currentHDPathOrCurve()[`curve`]
+      )
+      this.importedAddress = wallet.cosmosAddress
+      this.$store.commit(`updateField`, {
+        field: `HDPath`,
+        value: this.currentHDPathOrCurve()[`HDPath`],
+      })
+      this.$store.commit(`updateField`, {
+        field: `curve`,
+        value: this.currentHDPathOrCurve()[`curve`],
+      })
+    },
+    numberAttemptsController(networkCryptoTypes, attempt) {
+      if (attempt >= networkCryptoTypes.length) {
+        return attempt - networkCryptoTypes.length
+      } else {
+        return attempt
+      }
+    },
   },
   validations: () => ({
     name: { required, minLength: minLength(3), nameExists },
   }),
 }
 </script>
-<style scoped>
+<style>
 .address {
   word-break: break-all;
   font-size: 0.9rem;
   color: var(--txt);
 }
+
+.session-main p {
+  margin-bottom: 0;
+}
+
+.retry-link {
+  font-size: 12px;
+  cursor: pointer;
+  color: var(--link);
+}
+
+.retry-link:hover {
+  color: var(--link-hover);
+}
+
+.tm-form-group__field {
+  position: unset !important;
+}
+
+.algo { 
+  position: absolute;
```

Comment:
this shouldn't be absolute - it should flow naturally in the label area

Reply:
can you take care of this please